crates/ty_python_semantic/src/types/call/bind.rs

```diff
@@ -2304,6 +2334,20 @@ pub struct MatchedArgument {
     /// elements must have been successfully matched. (That means that this can be `false` while
     /// the `parameters` field is non-empty.)
     pub matched: bool,
+
+    /// The types of a variadic argument when it's unpacked. For example, given a `*args` whose
+    /// type is `tuple[A, B, C]`, this would be a 3 element vector containing the 3 types.
```

Comment:
It looks to me like the length of this vector is always the same as the length of the `parameters` vector -- that is, these are the types assigned to each matched parameter. This isn't necessarily the same as the number of types in the argument type (which might not be a fixed-length iterable). I think this comment could be clearer about this?

Reply:
Thanks, I've expanded the comment to include this and some additional notes around the fact that this is only populated for a variadic argument.

Reply:
I've also renamed the field from `types` to `unpacked_variadic_types`